File: app/src/main/java/com/sdex/activityrunner/util/CheckRootTask.java
package com.sdex.activityrunner.util;

import android.os.AsyncTask;

import eu.chainfire.libsuperuser.Shell;

/**
 * Author: Yuriy Mysochenko
 * Date: 05-Dec-17
 */

public class CheckRootTask extends AsyncTask<Void, Void, Integer> {

  public static final int ROOT_IS_NOT_AVAILABLE = 10;
  public static final int ACCESS_IS_NOT_GIVEN = 20;

  private final Callback callback;

  public CheckRootTask(Callback callback) {
    this.callback = callback;
  }

  @Override
  protected Integer doInBackground(Void[] params) {
    if (!Shell.SU.available()) {
      return ACCESS_IS_NOT_GIVEN;
    }
    return 0;
  }

  @Override
  protected void onPostExecute(Integer result) {
    super.onPostExecute(result);
    this.callback.onStatusChanged(result);
  }

  public interface Callback {

    void onStatusChanged(int status);
  }
}

File: app/src/main/java/com/sdex/activityrunner/util/CheckRootTask.kt
package com.sdex.activityrunner.util

import android.os.AsyncTask

class CheckRootTask(private val callback: Callback) : AsyncTask<Void, Void, Int>() {

  override fun doInBackground(params: Array<Void>): Int? {
    return if (RootUtils.isSuAvailable()) RESULT_OK else ACCESS_IS_NOT_GIVEN
  }

  override fun onPostExecute(result: Int?) {
    super.onPostExecute(result)
    this.callback.onStatusChanged(result!!)
  }

  interface Callback {

    fun onStatusChanged(status: Int)
  }

  companion object {

    const val RESULT_OK = 0
    const val ACCESS_IS_NOT_GIVEN = 1
  }
}
